### PR TITLE
Feature decor

### PR DESCRIPTION
**Pull request by raoulkent**
Obs! Most lines-of-code are small fixes to code style from https://github.com/Riddarvid/Gnomes-Tomes-and-Catacombs/commit/945e521ec873aa1036886afabf84cd6d4e5d94a8

### CODE REVIEW

File: game/core/src/main/java/com/gg/game/Model/Character.java [hunk 1]
```diff
@@ -14,28 +13,15 @@
         this.textureType = textureType;
     }
 
+    // TODO: Defensive copying?
     @Override
-    public void setPos(int x, int y) {
-        position = new Vector(x, y);
+    public TextureType getTexture() {
```
Comment (fourtooths) [inline game/core/src/main/java/com/gg/game/Model/Character.java]: Vad är TextureType?

File: game/core/src/main/java/com/gg/game/Model/Positionable.java [hunk 2]
```diff
@@ -1,7 +1,17 @@
 package com.gg.game.Model;
 
-public interface Positionable {
-    void setPos(int x, int y);
-    void setPos(Vector position);
-    Vector getPos();
-}
+abstract class Positionable {
```
Comment (fourtooths) [inline game/core/src/main/java/com/gg/game/Model/Positionable.java]: Föredrar delegering mer än arv.


```suggestion
interface Positionable {
    void setPos(int x, int y);
    void setPos(Vector position);
    Vector getPos();
}

class Decor implements positionable{
Vector vector = ....

@Override
parentMethod(){
vector.doStuff();
}
```

Comment (raoulkent) [reply game/core/src/main/java/com/gg/game/Model/Positionable.java]: Yeah I agree. The more I sat with it the more I like that idea. Will change that up before I make another PR.

File: game/core/src/main/java/com/gg/game/Model/Character.java (hunk 1 above)
Comment (raoulkent) [reply game/core/src/main/java/com/gg/game/Model/Character.java]: Currently an ENUM value right?
I'm not too fond of the solution if I'm honest.

**Pull request closed without merging**